### PR TITLE
add alpn option for UpstreamHttpProtocol enum

### PR DESCRIPTION
**Pull request by buildbreaker**
For: https://github.com/envoyproxy/envoy-mobile/issues/1548

Signed-off-by: Alan Chiu <achiu@lyft.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: add alpn option for UpstreamHttpProtocol
Risk Level: low
Testing: local
Docs Changes: n/a
Release Notes: n/a
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: library/swift/UpstreamHttpProtocol.swift [hunk 1]
```diff
@@ -21,6 +24,8 @@ public enum UpstreamHttpProtocol: Int, CaseIterable {
   /// - parameter stringValue: Case-insensitive string value to use for initialization.
   init(stringValue: String) {
     switch stringValue.lowercased() {
+    case "alpn"
```
Comment (rebello95) [inline library/swift/UpstreamHttpProtocol.swift:27]:
```suggestion
    case "alpn":
```

**Review by rebello95 — DISMISSED** [review]

File: library/swift/UpstreamHttpProtocol.swift (hunk 1 above)
Comment (buildbreaker) [reply library/swift/UpstreamHttpProtocol.swift:27]: lol is that why?!

Comment (rebello95) [reply library/swift/UpstreamHttpProtocol.swift]: ye

File: library/swift/UpstreamHttpProtocol.swift [hunk 2]
```diff
@@ -3,12 +3,15 @@ import Foundation
 /// Available upstream HTTP protocols.
 @objc
 public enum UpstreamHttpProtocol: Int, CaseIterable {
+  case alpn
```
Comment (rebello95) [inline library/swift/UpstreamHttpProtocol.swift:6]: docs for these might be useful - ALPN is probably not something people are familiar with if they don't work in networking much. Also, should we make this the default for requests or will that be a separate PR?

**Review by rebello95 — DISMISSED** [review]

File: library/swift/UpstreamHttpProtocol.swift (hunk 2 above)
Comment (buildbreaker) [reply library/swift/UpstreamHttpProtocol.swift:6]: Hm. Let's do both the defaulting and docs it in this PR too.

Comment (buildbreaker) [reply library/swift/UpstreamHttpProtocol.swift:6]: Im thinking of adding a section about the upstream protocols here: https://envoy-mobile.github.io/docs/envoy-mobile/latest/api/http.html

**Review by rebello95 — APPROVED** [review]

**Comment by junr03** [comment]
re-open if needed
